### PR TITLE
validate: fix panic

### PR DESCRIPTION
**Pull request by kolyshkin**
This is a carry of #688, using a slightly different fix. In addition, it adds a test case, and fixes the error texts.

Closes: https://github.com/opencontainers/runtime-tools/pull/688

### CODE REVIEW

**Review by tianon — APPROVED** [review]

**Comment by kolyshkin** [comment]
Need another LGTM on this one, @opencontainers/runtime-tools-maintainers PTAL

**Review by hqhq — APPROVED** [review]

**Pull request merged**
